### DISCUSSION TITLE
iOS/Android versions identification during build

### DISCUSSION THREAD

**Discussion by annunal** [beeware/toga#2378] [Q&A]
I am trying to maintain the same source for all versions but I understand that this is not possible for all the components. For example toga.DateInput does not exists in mac and so I need to use TextInput.

Is it possible to have compilation keywords in the code so that some parts are used for Android and other parts of code for mac or IoS and detect thecompilation flag in briefcase ?  A sort of preprocessing so that depending on the compilation for one or anothe rplatform sme parts are used ?

What is the best way to keep one source with different parts for each platform ?

Thanks for suggestions
Alessandro

**Comment by rmartin16** [comment]
You can use `toga.platform.current_platform` to arbitrate code execution for different platforms.

There's some example usage in the [screenshot example](https://github.com/beeware/toga/blob/main/examples/screenshot/screenshot/app.py).
